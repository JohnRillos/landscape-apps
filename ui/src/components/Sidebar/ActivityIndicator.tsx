import React from 'react';
import BulletIcon from '../icons/BulletIcon';

interface ActivityIndicatorProps {
  count: number;
}

export default function ActivityIndicator({ count }: ActivityIndicatorProps) {
  return (
    <div>
      {count === 0 ? (
        <BulletIcon className="h-6 w-6 rounded bg-gray-50 p-2" />
      ) : (
        <div className="flex h-6 w-6 items-center justify-center rounded bg-gray-50">
          {count > 99 ? '99+' : count}
        </div>
      )}
    </div>
  );
}
