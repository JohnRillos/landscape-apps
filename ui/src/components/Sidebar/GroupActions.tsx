import cn from 'classnames';
import React, { PropsWithChildren, useCallback, useState } from 'react';
import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
import { Link, useLocation } from 'react-router-dom';
import InviteIcon16 from '../icons/InviteIcon16';
import LinkIcon16 from '../icons/LinkIcon16';
import PinIcon16 from '../icons/PinIcon16';
import useCopyToClipboard from '../../logic/useCopyToClipboard';
import Person16Icon from '../icons/Person16Icon';
import EllipsisIcon from '../icons/EllipsisIcon';
import BulletIcon from '../icons/BulletIcon';
import { useBriefs } from '../../state/chat';

export function useGroupActions(flag: string) {
  const [_copied, doCopy] = useCopyToClipboard();
  const [isOpen, setIsOpen] = useState(false);
  const [copyItemText, setCopyItemText] = useState('Copy Group Link');

  const onCopyClick = useCallback(
    // eslint-disable-next-line prefer-arrow-callback
    function <T>(e: React.MouseEvent<T>) {
      e.stopPropagation();
      doCopy(flag);
    },
    [doCopy, flag]
  );

  const onCopySelect = useCallback((e: Event) => {
    e.preventDefault();
    setCopyItemText('Copied!');
    setTimeout(() => {
      setCopyItemText('Copy Group Link');
      setIsOpen(false);
    }, 1000);
  }, []);

  const onPinClick = useCallback(
    // eslint-disable-next-line prefer-arrow-callback
    function <T>(e: React.MouseEvent<T>) {
      e.stopPropagation();
      // TODO
      // eslint-disable-next-line no-console
      console.log('pin');
    },
    []
  );

  return {
    isOpen,
    setIsOpen,
    copyItemText,
    onCopySelect,
    onCopyClick,
    onPinClick,
  };
}

type GroupActionsProps = PropsWithChildren<{
  flag: string;
  className?: string;
}>;

export default function GroupActions({
  flag,
  className,
  children,
}: GroupActionsProps) {
  const location = useLocation();
  const briefs = useBriefs();
  const hasActivity = (briefs[flag]?.count ?? 0) > 0;

  const {
    isOpen,
    setIsOpen,
    copyItemText,
    onCopySelect,
    onCopyClick,
    onPinClick,
  } = useGroupActions(flag);

  return (
    <div className={className}>
      <DropdownMenu.Root open={isOpen} onOpenChange={setIsOpen}>
        <DropdownMenu.Trigger asChild>
          {children || (
            <div className="relative h-6 w-6">
              {!isOpen && hasActivity ? (
                <BulletIcon
                  className="absolute h-6 w-6 text-blue transition-opacity group-focus-within:opacity-0 group-hover:opacity-0"
                  aria-label="Has Activity"
                />
              ) : null}
              <button
                className={cn(
                  'default-focus absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 rounded-lg p-0.5 transition-opacity focus-within:opacity-100 hover:opacity-100 group-focus-within:opacity-100 group-hover:opacity-100',
                  hasActivity && 'text-blue',
                  isOpen ? 'opacity:100' : 'opacity-0'
                )}
                aria-label="Open Group Options"
              >
                <EllipsisIcon className="h-6 w-6" />
              </button>
            </div>
          )}
        </DropdownMenu.Trigger>
        <DropdownMenu.Content className="dropdown min-w-52 text-gray-800">
          <DropdownMenu.Item
            asChild
            className="dropdown-item rounded-none text-blue"
          >
            <Link
              to={`/groups/${flag}/invite`}
              state={{ backgroundLocation: location }}
              className="flex items-center space-x-2"
            >
              <InviteIcon16 className="h-6 w-6 opacity-60" />
              <span className="pr-2">Invite People</span>
            </Link>
          </DropdownMenu.Item>
          <DropdownMenu.Item
            className={
              'dropdown-item flex items-center space-x-2 rounded-none text-blue'
            }
            onClick={onCopyClick}
            onSelect={onCopySelect}
          >
            <LinkIcon16 className="h-6 w-6 opacity-60" />
            <span className="pr-2">{copyItemText}</span>
          </DropdownMenu.Item>
          <DropdownMenu.Item
            className="dropdown-item flex items-center space-x-2 rounded-none"
            onClick={onPinClick}
          >
            <PinIcon16 className="h-6 w-6 text-gray-600" />
            <span className="pr-2">Pin</span>
          </DropdownMenu.Item>
          <DropdownMenu.Item asChild className="dropdown-item rounded-none">
            <Link
              to={`/groups/${flag}/info`}
              state={{ backgroundLocation: location }}
              className="flex items-center space-x-2"
            >
              <Person16Icon className="m-1 h-4 w-4 text-gray-600" />
              <span className="pr-2">Members &amp; Group Info</span>
            </Link>
          </DropdownMenu.Item>
        </DropdownMenu.Content>
      </DropdownMenu.Root>
    </div>
  );
}
