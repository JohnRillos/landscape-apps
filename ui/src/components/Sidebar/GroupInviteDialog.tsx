import React, { useCallback } from 'react';
import Dialog, { DialogContent } from '../Dialog';

interface GroupInviteDialogProps {
  flag: string;
  onClose: () => void;
  onOpenChange: (open: boolean) => void;
  open: boolean;
}

export default function GroupInviteDialog({
  flag,
  onClose,
  onOpenChange,
  open,
}: GroupInviteDialogProps) {
  const onInvite = useCallback(() => onClose(), [onClose]); // TODO: invite poke

  return (
    <Dialog open={open} onOpenChange={onOpenChange}>
      <DialogContent containerClass="max-w-md" showClose>
        <div className="flex flex-col">
          <h2 className="mb-4 text-lg font-bold">Invite To Group</h2>
          <p className="mb-7 leading-5">Enter a username to invite to {flag}</p>
          <div className="flex items-center justify-end space-x-2">
            <button onClick={onClose} className="button" type="button">
              Cancel
            </button>

            <button
              onClick={onInvite}
              className="button bg-blue text-white"
              type="button"
            >
              Invite
            </button>
          </div>
        </div>
      </DialogContent>
    </Dialog>
  );
}
