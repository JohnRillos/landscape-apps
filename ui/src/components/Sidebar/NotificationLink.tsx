import React from 'react';
import { To } from 'react-router-dom';
import BulletIcon from '../icons/BulletIcon';
import SidebarLink from './SidebarLink';

interface NotificationLinkProps {
  count: number;
  title: string;
  to: To;
}

export default function NotificationLink({
  count,
  title,
  to,
}: NotificationLinkProps) {
  const notificationIcon = (
    <div>
      {count === 0 ? (
        <BulletIcon className="h-6 w-6 bg-gray-50 p-2" />
      ) : (
        <div className="flex h-6 w-6 items-center justify-center rounded bg-gray-50">
          {count > 99 ? '99+' : count}
        </div>
      )}
    </div>
  );

  return (
    <SidebarLink icon={notificationIcon} to={to}>
      {title}
    </SidebarLink>
  );
}
