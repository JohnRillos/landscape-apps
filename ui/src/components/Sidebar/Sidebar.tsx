import classNames from 'classnames';
import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { ModalLocationState } from '../../logic/routing';
import useMedia from '../../logic/useMedia';
import { useGangList, useGroup, useGroupList } from '../../state/groups';
import Divider from '../Divider';
import GangName from '../GangName/GangName';
import XIcon from '../icons/XIcon';
import SidebarLink from './SidebarLink';

function GroupItem({ flag }: { flag: string }) {
  const group = useGroup(flag);
  return (
    <SidebarLink to={`/groups/${flag}`} retainState>
      {group?.meta.title}
    </SidebarLink>
  );
}

function GangItem(props: { flag: string }) {
  const { flag } = props;
  return (
    <SidebarLink to={`/gangs/${flag}`}>
      <GangName flag={flag} />
    </SidebarLink>
  );
}

export default function Sidebar() {
  const groups = useGroupList();
  const gangs = useGangList();
  const location = useLocation();
  const isMobile = useMedia('(max-width: 639px)');
  const routeState = location.state as ModalLocationState | null;

  return (
    <nav className="h-full">
      <div
        className={classNames(
          'h-full min-w-56 border-r-2 border-gray-50 bg-white',
          isMobile && 'fixed top-0 left-0 z-50 w-full'
        )}
      >
        {isMobile ? (
          <header className="flex items-center border-b-2 border-gray-50 p-4">
            <h1 className="text-lg font-bold">Groups</h1>
            {routeState?.backgroundLocation ? (
              <Link
                to={routeState.backgroundLocation}
                className="icon-button ml-auto h-8 w-8"
                aria-label="Close Main Menu"
              >
                <XIcon className="h-6 w-6" />
              </Link>
            ) : null}
          </header>
        ) : null}
        <ul className="p-2">
          <SidebarLink to="/">Groups</SidebarLink>
          <SidebarLink to="/profile">Profile</SidebarLink>
          <SidebarLink to="/groups/new">New Group</SidebarLink>
          <SidebarLink to="/groups/join">Join Group</SidebarLink>
          <SidebarLink to="/dm">Direct Messages</SidebarLink>
          <Divider>All Groups</Divider>
          {groups.map((flag) => (
            <GroupItem key={flag} flag={flag} />
          ))}
          {gangs.length > 0 ? <Divider>Pending</Divider> : null}
          {gangs.map((flag) => (
            <GangItem key={flag} flag={flag} />
          ))}
        </ul>
      </div>
    </nav>
  );
}
