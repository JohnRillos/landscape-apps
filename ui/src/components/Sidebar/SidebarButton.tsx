import cn from 'classnames';
import React from 'react';

interface SidebarProps extends React.ComponentPropsWithoutRef<'button'> {
  className?: string;
  color?: string;
  icon?: React.ReactNode;
}

export default function SidebarButton({
  children,
  className,
  color = 'text-gray-600',
  icon,
  ...rest
}: SidebarProps) {
  return (
    <li>
      <button
        className={cn(
          'default-focus flex w-full items-center space-x-3 rounded-lg p-2 text-lg font-semibold hover:bg-gray-50 sm:text-base',
          color,
          className
        )}
        {...rest}
      >
        {icon ? (
          icon
        ) : (
          <div className="h-6 w-6 rounded border-2 border-gray-100" />
        )}
        {typeof children === 'string' ? <h3>{children}</h3> : children}
      </button>
    </li>
  );
}
