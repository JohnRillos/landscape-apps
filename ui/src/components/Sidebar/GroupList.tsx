import cn from 'classnames';
import React, { PropsWithChildren, useEffect } from 'react';
import { uniq, without } from 'lodash';
import { DndProvider, useDrag, useDrop } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import { TouchBackend } from 'react-dnd-touch-backend';
import * as Popover from '@radix-ui/react-popover';
import { useIsMobile } from '@/logic/useMedia';
import {
  useGang,
  useGangList,
  useGroup,
  useGroupState,
} from '@/state/groups/groups';
import { SettingsState, useSettingsState } from '@/state/settings';
import GroupAvatar from '@/groups/GroupAvatar';
import GroupActions from '@/groups/GroupActions';
import { Group } from '@/types/groups';
import Divider from '../Divider';
import useNavStore from '../Nav/useNavStore';
import SidebarItem from './SidebarItem';

const dragTypes = {
  GROUP: 'group',
};

const selOrderedPins = (s: SettingsState) => ({
  order: s.groups.orderedGroupPins,
  loaded: s.loaded,
});

function DraggableGroupItem({ flag }: { flag: string }) {
  const group = useGroup(flag);
  const navPrimary = useNavStore((state) => state.navigatePrimary);

  const [{ isDragging }, drag] = useDrag(() => ({
    type: dragTypes.GROUP,
    item: { flag },
    collect: (monitor) => ({
      isDragging: !!monitor.isDragging(),
    }),
  }));

  return (
    <div
      ref={drag}
      className={cn(
        'absolute w-full',
        isDragging ? 'opacity-0' : 'opacity-100'
      )}
    >
      <SidebarItem
        div
        icon={<GroupAvatar size="h-12 w-12 sm:h-6 sm:w-6" {...group?.meta} />}
        actions={<GroupActions flag={flag} />}
        onClick={() => navPrimary('group', flag)}
      >
        {group?.meta.title}
      </SidebarItem>
    </div>
  );
}

function GroupItem({ flag }: { flag: string }) {
  const group = useGroup(flag);
  const navPrimary = useNavStore((state) => state.navigatePrimary);

  return (
    <SidebarItem
      icon={<GroupAvatar size="h-12 w-12 sm:h-6 sm:w-6" {...group?.meta} />}
      actions={<GroupActions flag={flag} />}
      onClick={() => navPrimary('group', flag)}
    >
      {group?.meta.title}
    </SidebarItem>
  );
}

function GroupItemContainer({
  flag,
  children,
}: PropsWithChildren<{ flag: string }>) {
  const { order } = useSettingsState(selOrderedPins);
  const [{ isOver }, drop] = useDrop<
    { flag: string },
    undefined,
    { isOver: boolean }
  >(
    () => ({
      accept: dragTypes.GROUP,
      drop: ({ flag: itemFlag }) => {
        if (!itemFlag || itemFlag === flag) {
          return undefined;
        }
        // [1, 2, 3, 4] 1 -> 3
        // [2, 3, 4]
        const beforeSlot = order.indexOf(itemFlag) < order.indexOf(flag);
        const orderWithoutOriginal = without(order, itemFlag);
        const slicePoint = orderWithoutOriginal.indexOf(flag);
        // [2, 3] [4]
        const left = orderWithoutOriginal.slice(
          0,
          beforeSlot ? slicePoint + 1 : slicePoint
        );
        const right = orderWithoutOriginal.slice(slicePoint);
        // concat([2, 3], [1], [4])
        const newOrder = uniq(left.concat([itemFlag], right));
        // [2, 3, 1, 4]
        useSettingsState
          .getState()
          .putEntry('groups', 'orderedGroupPins', newOrder);

        return undefined;
      },
      collect: (monitor) => ({
        isOver: !!monitor.isOver(),
      }),
    }),
    [flag]
  );

  return (
    <li
      ref={drop}
      className={cn(
        'relative flex h-16 w-full ring-4 sm:h-10',
        isOver && 'ring-blue-500',
        !isOver && 'ring-transparent'
      )}
    >
      {children}
    </li>
  );
}

// Gang is a pending group invite
function GangItem(props: { flag: string }) {
  const { flag } = props;
  const { preview, claim } = useGang(flag);
  const isMobile = useIsMobile();

  const handleCancel = async () => {
    await useGroupState.getState().reject(flag);
  };

  if (!claim) {
    return null;
  }

  return (
    <Popover.Root>
      <Popover.Anchor>
        <Popover.Trigger asChild>
          <SidebarItem
            icon={
              <GroupAvatar
                {...preview?.meta}
                size="h-12 w-12 sm:h-6 sm:w-6"
                className="opacity-60"
              />
            }
          >
            <span className="inline-block w-full truncate opacity-60">
              {preview ? preview.meta.title : flag}
            </span>
          </SidebarItem>
        </Popover.Trigger>
      </Popover.Anchor>
      <Popover.Content
        side={isMobile ? 'top' : 'right'}
        sideOffset={isMobile ? 0 : 16}
      >
        <div className="flex w-[200px] flex-col space-y-4 rounded-lg bg-white p-4 leading-5 drop-shadow-lg">
          <span>You are currently joining this group.</span>
          <span>
            It may take a few minutes depending on the host&apos;s and your
            connection.
          </span>
          <div className="flex">
            <Popover.Close>
              <button
                className="small-button bg-gray-50 text-gray-800"
                onClick={handleCancel}
              >
                Cancel Join
              </button>
            </Popover.Close>
          </div>
        </div>
      </Popover.Content>
    </Popover.Root>
  );
}

interface GroupListProps {
  className?: string;
  pinned?: boolean;
  groups: [string, Group][];
  pinnedGroups: [string, Group][];
}

export default function GroupList({
  className,
  pinned = false,
  groups,
  pinnedGroups,
}: GroupListProps) {
  const isMobile = useIsMobile();
  const gangs = useGangList();
  const { order, loaded } = useSettingsState(selOrderedPins);

  useEffect(() => {
    const hasKeys = order && !!order.length;
    const pinnedKeys = Object.keys(pinnedGroups);
    const hasPinnedKeys = pinnedKeys.length > 0;

    if (!loaded) {
      return;
    }

    // Correct order state, fill if none, remove duplicates, and remove
    // old uninstalled app keys
    if (!hasKeys && hasPinnedKeys) {
      useSettingsState
        .getState()
        .putEntry('groups', 'orderedGroupPins', pinnedKeys);
    } else if (order.length < pinnedKeys.length) {
      useSettingsState
        .getState()
        .putEntry('groups', 'orderedGroupPins', uniq(order.concat(pinnedKeys)));
    } else if (order.length > pinnedKeys.length && hasPinnedKeys) {
      useSettingsState
        .getState()
        .putEntry(
          'groups',
          'orderedGroupPins',
          uniq(order.filter((key) => key in pinnedGroups).concat(pinnedKeys))
        );
    }
  }, [pinnedGroups, order, loaded]);

  return pinned ? (
    <DndProvider
      backend={isMobile ? TouchBackend : HTML5Backend}
      options={
        isMobile
          ? {
              delay: 50,
              scrollAngleRanges: [
                { start: 30, end: 150 },
                { start: 210, end: 330 },
              ],
            }
          : undefined
      }
    >
      <li className="-mb-2 flex items-center sm:m-0 md:pr-2">
        <Divider>Pinned</Divider>
        <div className="grow border-b-2 border-gray-100" />
      </li>
      {pinnedGroups.map(([flag]) => (
        <GroupItemContainer flag={flag} key={flag}>
          <DraggableGroupItem flag={flag} />
        </GroupItemContainer>
      ))}
    </DndProvider>
  ) : (
    <ul className={cn('h-full space-y-3 p-2 sm:space-y-0', className)}>
      {gangs.map((flag) => (
        <GangItem key={flag} flag={flag} />
      ))}
      {groups
        .filter(
          ([flag, _group]) => !pinnedGroups.map(([f, _]) => f).includes(flag)
        )
        .map(([flag]) => (
          <GroupItem key={flag} flag={flag} />
        ))}
    </ul>
  );
}
