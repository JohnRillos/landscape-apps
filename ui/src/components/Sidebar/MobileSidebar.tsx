import React from 'react';
import useNavStore from '../Nav/useNavStore';
import useSidebarSort from '../../logic/useSidebarSort';
import SidebarSorter from './SidebarSorter';
import NavTab from '../NavTab';
import GroupIcon from '../icons/GroupIcon';
import ActivityIndicator from './ActivityIndicator';
import AsteriskIcon from '../icons/Asterisk16Icon';
import MagnifyingGlassIcon from '../icons/MagnifyingGlassIcon';
import GroupList from './GroupList';
import { usePinnedGroups } from '../../state/groups/groups';

export default function MobileSidebar() {
  const secondary = useNavStore((state) => state.secondary);
  const pinned = usePinnedGroups();
  const { sortFn, setSortFn, sortOptions } = useSidebarSort();
  // TODO: get notification count from hark store
  const notificationCount = 0;

  return (
    <section className="fixed inset-0 z-40 flex h-full w-full flex-col border-r-2 border-gray-50 bg-white">
      <header className="flex-none px-2 py-1">
        {pinned ? (
          <ul>
            <GroupList pinned />
          </ul>
        ) : null}
        {secondary === 'main' ? (
          <SidebarSorter
            sortFn={sortFn}
            setSortFn={setSortFn}
            sortOptions={sortOptions}
            isMobile={true}
          />
        ) : (
          <h1 className="p-2 text-xl font-medium">
            {secondary === 'notifications'
              ? 'Notifications'
              : secondary === 'search'
              ? 'Search Groups'
              : null}
          </h1>
        )}
      </header>
      <nav className="h-full flex-1 overflow-y-auto">
        {secondary === 'main' ? (
          <GroupList />
        ) : secondary === 'notifications' ? (
          <div />
        ) : secondary === 'search' ? (
          <div />
        ) : null}
      </nav>
      <footer className="flex-none border-t-2 border-gray-50">
        <nav>
          <ul className="flex items-center">
            <NavTab loc="main">
              <GroupIcon className="mb-0.5 h-6 w-6" />
              Groups
            </NavTab>
            <NavTab loc="notifications">
              <ActivityIndicator count={notificationCount} className="mb-0.5" />
              Notifications
            </NavTab>
            <NavTab loc="search">
              <MagnifyingGlassIcon className="mb-0.5 h-6 w-6" />
              Search
            </NavTab>
            <a
              className="no-underline"
              href="https://github.com/tloncorp/homestead/issues/new?assignees=&amp;labels=bug&amp;template=bug_report.md&amp;title=groups:"
              target="_blank"
              rel="noreferrer"
            >
              <NavTab>
                <AsteriskIcon className="mb-0.5 h-6 w-6" />
                Submit Issue
              </NavTab>
            </a>
          </ul>
        </nav>
      </footer>
    </section>
  );
}
