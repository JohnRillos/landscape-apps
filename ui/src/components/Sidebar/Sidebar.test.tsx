/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Sidebar from './Sidebar';


describe('Sidebar', () => {
  it('renders as expected', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Sidebar />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
