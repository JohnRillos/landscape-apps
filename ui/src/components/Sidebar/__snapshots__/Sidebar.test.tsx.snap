// Vitest Snapshot v1

exports[`Sidebar > renders as expected 1`] = `
<DocumentFragment>
  <nav
    class="flex h-full w-64 flex-col bg-white"
  >
    <ul
      class="flex w-full flex-col px-2 pt-2"
    >
      <li
        class="group relative flex w-full items-center justify-between rounded-lg text-lg font-semibold sm:text-base text-gray-600 hover:bg-gray-50"
      >
        <a
          class="default-focus flex w-full flex-1 items-center space-x-3 rounded-lg p-2 font-semibold"
          href="/profile/edit"
        >
          <div
            class="relative flex flex-none items-center justify-center rounded bg-black w-6 h-6 rounded p-1.5"
            style="background-color: rgb(0, 0, 0);"
          >
            <svg
              class=""
              style="display: block; width: 12px; height: 12px;"
              version="1.1"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#000000"
                height="12px"
                width="12px"
                x="0px"
                y="0px"
              />
              <g
                transform="matrix(0.046875,0,0,0.046875,0,0)"
              >
                <g
                  fill="none"
                  transform="matrix(1,0,0,1,0,0)"
                >
                  <rect
                    dataisgeon="true"
                    fill="white"
                    height="128"
                    stroke="#000000"
                    stroke-linecap="square"
                    stroke-width="0.8px"
                    vector-effect="non-scaling-stroke"
                    width="128"
                  />
                </g>
                <g
                  fill="none"
                  transform="matrix(1,0,0,1,128,0)"
                >
                  <circle
                    cx="64"
                    cy="64"
                    dataisgeon="true"
                    fill="white"
                    r="64"
                    stroke="#000000"
                    stroke-linecap="square"
                    stroke-width="0.8px"
                    vector-effect="non-scaling-stroke"
                  />
                </g>
                <g
                  fill="none"
                  transform="matrix(1,0,0,1,0,128)"
                >
                  <path
                    d="M0.0541 0C70.7217 0.0292317 128 57.3256 128 128C57.3177 128 0.0164917 70.7089 7.62806e-06 0.0305091C7.62851e-06 0.0203397 -4.44317e-10 0.01017 0 0H0.0541Z"
                    dataisgeon="true"
                    fill="white"
                    stroke="#000000"
                    stroke-linecap="square"
                    stroke-width="0.8px"
                    vector-effect="non-scaling-stroke"
                  />
                </g>
                <g
                  fill="none"
                  transform="matrix(1,0,0,1,128,128)"
                >
                  <path
                    d="M64 128H0L5.59506e-06 0L64 5.59506e-06C99.3462 8.68512e-06 128 28.6538 128 64C128 99.3462 99.3462 128 64 128Z"
                    dataisgeon="true"
                    fill="white"
                    stroke="#000000"
                    stroke-linecap="square"
                    stroke-width="0.8px"
                    vector-effect="non-scaling-stroke"
                  />
                </g>
              </g>
            </svg>
          </div>
          <div
            class="max-w-full flex-1 truncate text-left"
          >
            <span>
              <span
                aria-hidden="true"
              >
                ~
              </span>
              <span>
                finned
              </span>
              <span
                aria-hidden="true"
              >
                -
              </span>
              <span
                aria-hidden="false"
              >
                palmer
              </span>
            </span>
          </div>
        </a>
      </li>
      <li
        class="group relative flex w-full items-center justify-between rounded-lg text-lg font-semibold sm:text-base text-gray-600 hover:bg-gray-50"
      >
        <a
          class="default-focus flex w-full flex-1 items-center space-x-3 rounded-lg p-2 font-semibold"
          href="/notifications"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded text-sm font-semibold bg-gray-100"
          >
            <svg
              class="h-6 w-6"
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                class="fill-current"
                height="8"
                rx="4"
                width="8"
                x="8"
                y="8"
              />
            </svg>
          </div>
          <div
            class="max-w-full flex-1 truncate text-left"
            title="Notifications"
          >
            Notifications
          </div>
        </a>
      </li>
      <li
        class="group relative flex w-full items-center justify-between rounded-lg text-lg font-semibold sm:text-base text-gray-600 hover:bg-gray-50"
      >
        <a
          class="default-focus flex w-full flex-1 items-center space-x-3 rounded-lg p-2 font-semibold"
          href="/groups/find"
        >
          <svg
            class="m-1 h-4 w-4"
            fill="none"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="fill-current"
              clip-rule="evenodd"
              d="M6.00003 7c0-1.65685 1.34314-3 3-3C10.6569 4 12 5.34315 12 7s-1.3431 3-2.99997 3c-1.65686 0-3-1.34315-3-3Zm3-5c-2.76142 0-5 2.23858-5 5 0 1.01907.30487 1.96694.82838 2.75737l-.53552.53553-2 2c-.39052.3905-.39052 1.0237 0 1.4142.39053.3905 1.02369.3905 1.41422 0l2-2 .53551-.5355c.79044.5235 1.73832.8284 2.75741.8284C11.7615 12 14 9.76142 14 7s-2.2385-5-4.99997-5Z"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="max-w-full flex-1 truncate text-left"
          >
            <div
              class="flex items-center"
            >
              Find Groups
            </div>
          </div>
        </a>
      </li>
      <li
        class="group relative flex w-full items-center justify-between rounded-lg text-lg font-semibold sm:text-base text-blue hover:bg-blue-soft hover:dark:bg-blue-900"
      >
        <a
          class="default-focus flex w-full flex-1 items-center space-x-3 rounded-lg p-2 font-semibold"
          href="/groups/new"
        >
          <svg
            class="m-1 h-4 w-4"
            fill="none"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="stroke-current"
              d="M3 8h10M8 3v10"
              stroke-linecap="round"
              stroke-width="2"
            />
          </svg>
          <div
            class="max-w-full flex-1 truncate text-left"
            title="Create Group"
          >
            Create Group
          </div>
        </a>
      </li>
      <a
        class="no-underline"
        href="https://github.com/tloncorp/homestead/issues/new?assignees=&labels=bug&template=bug_report.md&title=groups:"
        rel="noreferrer"
        target="_blank"
      >
        <li
          class="group relative flex w-full items-center justify-between rounded-lg text-lg font-semibold sm:text-base text-yellow-600 dark:text-yellow-500 hover:bg-yellow-soft hover:dark:bg-yellow-800"
        >
          <button
            class="default-focus flex w-full flex-1 items-center space-x-3 rounded-lg p-2 font-semibold"
          >
            <svg
              class="m-1 h-4 w-4"
              fill="none"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="stroke-current"
                d="m8 8 4-4M8 8 4 4m4 4-4 4m4-4 4 4M8 8h5M8 8V3m0 5H3m5 0v5"
                stroke-linecap="round"
                stroke-width="2"
              />
            </svg>
            <div
              class="max-w-full flex-1 truncate text-left"
              title="Submit Issue"
            >
              Submit Issue
            </div>
          </button>
        </li>
      </a>
      <li
        class="p-2"
      >
        <button
          aria-haspopup="menu"
          aria-label="Groups Sort Options"
          class="default-focus flex items-center rounded-lg bg-gray-50 py-2 px-4 text-base font-semibold"
          data-state="closed"
          id="radix-0"
          type="button"
        >
          <span
            class="mr-2 pl-1"
          >
            All Groups: A â†’ Z
          </span>
          <svg
            class="h-4 w-4 text-gray-400"
            fill="none"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="stroke-current"
              d="m4 6 4 4 4-4"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </li>
    </ul>
    <ul
      class="h-full space-y-3 p-2 sm:space-y-0 flex-1 overflow-x-hidden overflow-y-scroll pr-0 pt-0"
    />
  </nav>
</DocumentFragment>
`;
